extensions/amp-inputmask/validator-amp-inputmask.protoascii

```diff
@@ -37,6 +37,8 @@ tags: {
     name: "mask"
     mandatory: true
     dispatch_key: NAME_DISPATCH
+    # Only allow named masks to appear alone in the mask list.
+    blacklisted_value_regex: "([^\\s]+\\scredit|credit\\s[^\\s]+)"
```

Comment:
What are you trying to disallow? Any text with space before or after `credit`? Is there a doc I can read that describes this more?

Reply:
The mask attribute specifies the masks to use for input masking behavior. If there are custom masks, there can be multiple passed in to allow users to enter more than one format. If a named mask is used, it may be the only one passed in.

I am trying to disallow multiple named masks with this regex (right now `credit` is the only one). I am using this regex to detect if multiple named masks are used. If there is the word credit before or after a space character with more text past it, we know that multiple masks are present in the mask attribute and should be disallowed.

Please let me know if there's a better way to do this, I wrote this up kind of quickly.

The [validator-amp-inputmask.out](https://github.com/ampproject/amphtml/pull/20347/files#diff-055a1bd2573139a24c030a533ce1a47d) shows what I am trying to make invalid with this regex. [go/amp-form-masking](https://docs.google.com/document/d/1tCw8oXTRuVTggA-0pXFQeLTJUS7qX8tlNY6RmhLRGZ4/edit#heading=h.sqt8ry69qn6i) is the main design doc.

Reply:
Discussed offline. It would be good to explode the regex with explanation of what each part means. E.g.

```
blacklisted_value_regex: "([^\\s]+"  # One or more non whitespace character
                         "\\s"       # A single whitespace
                         "credit"    # named mask
                         "|"         # or
                         "credit"    # named mask
                         ...
```

Reply:
Done.